src/io.jl

```diff
@@ -629,6 +621,7 @@ end
 
 struct NCReader
     dataset::CFDataset
+    dataset_times::Vector{DateTime}
```

Comment:
If I remember correctly NCDatasets doesn not always return a `Vector{DateTime}` for `dataset["time"][:]`, but, depending on the calendar used, other time types from CFTime.jl. We'd need to test this with a netCDF with such a calender though. I'd imagine we'd need `struct NCReader{T}` with `dataset_times::Vector{T}`.

Reply:
Yes, good point! Indeed, other time types are possible with NCDatasets depending on the calendar used. Suggested changes are in commit https://github.com/Deltares/Wflow.jl/pull/240/commits/9f02023b088ba5bcf92b49d80ed5e9af367c601f.